core/tests/test_utils.py

```diff
@@ -1811,6 +1811,28 @@ def process_and_flush_pending_tasks(self, queue_name=None):
             for queue in queue_names:
                 self.taskqueue_stub.FlushQueue(queue)
 
+    def _run_but_do_not_flush_pending_tasks(self):
```

Comment:
Remember, don't repeat code -- try to have process_and_flush_pending_tasks() above make use of this function somehow.

Reply:
Hmm I am trying but this seems kinda odd. The whole loop in `_run_but_do_not_flush_pending_tasks` should be under `while tasks` but some other functions should be outside the while loop such as the first `FlushQueue`(line 1783). Can you help me in putting _run_but_do_not_flush_pending_tasks inside process_and_flush_pending_tasks? Thanks!

Reply:
There are probably several solutions here. One other way you could avoid code duplication is to extract the "for task in tasks .... raise RuntimeError('MapReduce task ...')" bit into a private _execute_tasks() method -- then, both these methods can call it. 

Would something along those lines work?

Reply:
Done